api/lib/source/cosmosV0-source.js

```diff
@@ -224,14 +227,113 @@ class CosmosV0API extends RESTDataSource {
       validator.signing_info = signingInfos[consensusAddress]
     })
 
-    return validators.map((validator) =>
+    const validatorsWithoutProfiles = validators.map((validator) =>
       this.reducers.validatorReducer(
         this.network.id,
         signedBlocksWindow,
         validator,
         annualProvision
       )
     )
+    if (profile) {
+      return { validators, validatorsWithoutProfiles }
+    } else {
+      return validatorsWithoutProfiles
+    }
+  }
+
+  async getProfilesForValidators(
+    primitiveValidators,
+    validatorsWithoutProfiles,
+    fiatCurrency = 'USD'
+  ) {
+    // const networkList = await db.getNetworks()
+    validatorsWithoutProfiles = this.getRanksForValidators(
+      validatorsWithoutProfiles
+    )
+    return await Promise.all(
+      validatorsWithoutProfiles.map(async (validator) => {
+        const [
+          validatorProfile,
+          // latestValidatorNotifications,
+          allValidatorDelegations
+        ] = await Promise.all([
+          this.db.getValidatorProfile(validator.operatorAddress),
+          // this.db.getAccountNotifications(
+          //   validator.operatorAddress,
+          //   this.network.id
+          // ),
+          this.getAllValidatorDelegations(validator)
+        ])
+        const primitiveValidator = primitiveValidators.find(
+          ({ operator_address }) =>
+            operator_address === validator.operatorAddress
+        )
+        const fiatValuesResponse = await this.fiatValuesAPI.calculateFiatValues(
+          [{ amount: validator.tokens, denom: this.network.stakingDenom }],
+          fiatCurrency
+        )
+        const totalStakedAssets = fiatValuesResponse[this.network.stakingDenom]
+        return this.reducers.validatorProfileReducer(
+          validator,
+          primitiveValidator,
+          validatorProfile,
+          totalStakedAssets,
+          allValidatorDelegations.length,
+          this.network
+          // latestValidatorNotifications.map((notification) =>
```

Comment:
delete?

Reply:
see comment above

Reply:
don't leave comments in PRs bound for production